### PR TITLE
MBS-11078 / MBS-11081: Fix pregap bugs in Add medium

### PR DESCRIPTION
**Pull request by reosarevok**
### Fix MBS-11078 / MBS-11081

One of these (MBS-11078) is a newly introduced regression. The other (MBS-11081) has been there from the beginning, but hey, might as well fix it too so once the edit works again, it's also complete.

MBS-11081 also brings the icon to a couple other places related to discIDs, where it certainly won't hurt.

### CODE REVIEW

File: root/edit/details/AddMedium.js [hunk 1]
```diff
@@ -73,7 +73,7 @@ const CondensedTrackACs = ({
   }
 
   let thisCredit;
-  let thisPosition = 0;
+  let thisPosition = tracks[0].position - 1;
```
Comment (reosarevok) [inline root/edit/details/AddMedium.js:76]: Starting at thisPosition = 0 and adding 1 before comparison broke for stuff where the position *is* supposed to be 0. Now it should be safe.

Comment (mwiencek) [reply root/edit/details/AddMedium.js:76]: Will that continue to work once mediums can have an empty tracklist? We bail out if `!tracks`, but not if `!tracks?.length`.

Comment (reosarevok) [reply root/edit/details/AddMedium.js:76]: We shouldn't get there once https://github.com/metabrainz/musicbrainz-server/pull/1633 gets merged AFAICT.

**Review by mwiencek — APPROVED** [review]

**Review by yvanzo — APPROVED** [review]
:shipit:

File: root/static/scripts/common/components/PregapTrackIcon.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import * as React from 'react';
+
+const PregapTrackIcon = (): React.Element<'div'> => (
+  <div
+    className="pregap-track icon img"
+    title={l('This track is hidden in the pregap.')}
+  />
+);
+
+export default PregapTrackIcon;
```
Comment (yvanzo) [inline root/static/scripts/common/components/PregapTrackIcon.js:19]: That should probably be a separate commit. I found this macro conversion to be made in pull request https://github.com/metabrainz/musicbrainz-server/pull/1660 (commit https://github.com/metabrainz/musicbrainz-server/pull/1660/commits/17ef466f4caabbbf4db4f19c6392f16c36a41131#diff-8ea75b5338e161bca6ed99b878adb716R847) too.

Comment (reosarevok) [reply root/static/scripts/common/components/PregapTrackIcon.js:19]: Split the commit :)

Comment (yvanzo) [reply root/static/scripts/common/components/PregapTrackIcon.js:19]: Thanks!
